app.py

```diff
@@ -134,6 +147,7 @@ def login():
 def register():
     """Register a user."""
     data = request.form
+    # TODO we might need also to check if data has username or password
```

Comment:
if either the password or username is not set correctly the add_user call will return with an error, which is handled gracefully.